contracts/FindMarketCut.cdc

```diff
@@ -0,0 +1,88 @@
+import FindMarketCutStruct from "../contracts/FindMarketCutStruct.cdc"
+import FindMarketCutInterface from "../contracts/FindMarketCutInterface.cdc"
+import FindUtils from "../contracts/FindUtils.cdc"
+
+pub contract FindMarketCut {
+
+	pub let categoryToContractName: {String : String}
+
+	pub event Cut(tenant: String, type: String, category: String, contractName: String, cutInfo: [FindMarketCutStruct.EventSafeCut], action: String, remark: String?)
+
+	pub fun getRuleId(listingType: Type, nftType:Type, ftType:Type) : String {
+		let s : [String] = [listingType.identifier, nftType.identifier, ftType.identifier]
+		return FindUtils.joinString(s, sep: "-")
+	}
+
+	pub fun getCuts(tenant: String, listingType: Type, nftType:Type, ftType:Type): {String : FindMarketCutStruct.Cuts} {
```

Comment:
getCuts loops all the registered contracts and get the required cut under unique listing instance (ruleId) and return in a map of {category : requiredCuts}